packages/components/src/components/files/FilesListingForm.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from 'react';
+import { Button } from 'react-bootstrap';
+import { Map, Set, List } from 'immutable';
+import { IFile } from "./models";
+import { FileAttachmentForm } from "./FileAttachmentForm";
+import { FilesListing } from './FilesListing';
+
+interface Props {
+    files?: List<IFile>
+    headerText?: string
+    readOnlyFiles?: List<IFile>
+    readOnlyHeaderText?: string
+    addFileText?: string
+    noFilesMessage?: string
+    noReadOnlyFilesMessage?: string
+
+    handleUpload?: (files: Map<string, File>, cb: () => any) => any
+    handleDelete?: (file: string) => any
+    handleDownload?: (files: Set<string>) => any
+    canInsert?: boolean
+    canDelete?: boolean
+    useFilePropertiesEditTrigger?: boolean
+    onPropertyUpdate?: () => any
+    onSubmit?: () => any
+    onUploadFiles?: () => any
+    getFilePropertiesEditTrigger?: (file :IFile) => React.ReactNode
+}
+
+interface State {
+    selectedFiles?: Set<string>
+    confirmDeletionSet: Set<string>
+    showFileUploadPanel?: boolean
+}
+
+export class FilesListingForm extends React.Component<Props, State> {
+
+    static defaultProps = {
+        addFileText: "Attach File"
+    };
+
+    constructor(props: Props) {
+        super(props);
+
+        this.downloadSelectedFiles = this.downloadSelectedFiles.bind(this);
+        this.onFinishUploadFiles = this.onFinishUploadFiles.bind(this);
+        this.toggleUploadSection = this.toggleUploadSection.bind(this);
+        this.toggleFileSelection = this.toggleFileSelection.bind(this);
+        this.uploadAttachedFiles = this.uploadAttachedFiles.bind(this);
```

Comment:
Could remove these and change the functions to us the arrow notation, but I assume this was just brought along with your refactor of this component.